source/CommonJobs/CommonJobs.Application.EvalForm/Commands/UpdateCalificationsCommand.cs

```diff
@@ -0,0 +1,126 @@
+﻿using CommonJobs.Application.EvalForm.DTOs;
+using CommonJobs.Domain.Evaluations;
+using CommonJobs.Infrastructure.RavenDb;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace CommonJobs.Application.EvalForm.Commands
+{
+    public class UpdateCalificationsCommand : Command
+    {
+        private const string COMPANY = "_company";
+
+        private UpdateEvaluationDTO _updateEvaluation;
+
+        private string _loggedUser;
+
+        public UpdateCalificationsCommand(UpdateEvaluationDTO updateEvaluation, string loggedUser)
+        {
+            _updateEvaluation = updateEvaluation;
+
+            _loggedUser = loggedUser;
+        }
+
+        public override void Execute()
+        {
+            bool updateEvaluationComments = false;
+            bool updateEvaluationProject = false;
+
+            var storedEvaluation = RavenSession.Load<EmployeeEvaluation>(_updateEvaluation.EvaluationId);
+            if (storedEvaluation == null)
+            {
+                throw new ApplicationException(string.Format("Error: Evaluación {0} inexistente", _updateEvaluation.EvaluationId));
+            }
+
+            foreach (var calification in _updateEvaluation.Califications)
+            {
+                var storedCalification = RavenSession.Load<EvaluationCalification>(calification.CalificationId);
+                if (storedCalification == null)
+                {
+                    throw new ApplicationException(string.Format("Error: Calificación {0} inexistente para evaluación {1}", calification.CalificationId, _updateEvaluation.EvaluationId));
+                }
+
+                // Checks if the user trying to update the califications can actually do it
+                if (CanUpdate(_loggedUser, storedEvaluation, storedCalification))
+                {
+                    UpdateCalification(calification, storedCalification, _updateEvaluation.Finished);
```

Comment:
I have no issues with the update, it is something expected and, I am not sure, but it could be optimized and save when the unit of work is closed. Said that, I think that it would be nice to avoid SELECT+1 ([line 39](https://github.com/CommonJobs/CommonJobs/pull/387/files#diff-e00df1392631659f1018fc0f93e90667R39)).

Since I am not sure how to fix it, and today the RC should be released, I prefer to add a TODO mark in the code.